src/lib/MVC/Symfony/Controller/Content/PreviewController.php

```diff
@@ -187,6 +202,41 @@ protected function getForwardRequest(
             $forwardRequestParameters
         );
     }
+
+    /**
+     * @throws \Ibexa\Core\Base\Exceptions\BadStateException
+     */
+    private function buildResponseForGenericPreviewError(Location $location, Content $content, Exception $e): Response
+    {
+        $message = '';
+        try {
+            if ($location->isDraft() && $this->controllerChecker->usesCustomController($content, $location)) {
+                $message = <<<EOF
+<p>The view that rendered this location draft uses a custom controller, and resulted in a fatal error.</p>
+<p>Location View is deprecated, as it causes issues with preview, such as an empty location id when previewing the first version of a content.</p>
+EOF;
```

Comment:
Since PHP 7.3 those can be indented accordingly.

https://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.heredoc

```suggestion
                $message = <<<EOF
                    <p>The view that rendered this location draft uses a custom controller, and resulted in a fatal error.</p>
                    <p>Location View is deprecated, as it causes issues with preview, such as an empty location id when previewing the first version of a content.</p>
                    EOF;
```